### PR TITLE
Migrate Email Connector to JDK 11

### PR DESCRIPTION
**Pull request by Maninda**
## Purpose
JDK 11 migration. Resolves https://github.com/ballerina-platform/module-ballerina-email/issues/23

### CODE REVIEW

File: email-test-utils/src/main/java/module-info.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module io.ballerina.stdlib.email.testutils {
```
Comment (kalaiyarasiganeshalingam) [inline email-test-utils/src/main/java/module-info.java:1]: Missing license header

File: build.gradle [hunk 2]
```diff
@@ -163,13 +168,14 @@ subprojects {
...
-        compile group: 'org.testng', name: 'testng', version: "${testngVersion}"
-        compile group: 'org.jvnet.mimepull', name: 'mimepull', version: "${mimepullVersion}"
+//        compile group: 'org.ballerinalang', name: 'ballerina-lang', version: "${ballerinaLangVersion}"
+//        compile group: 'org.ballerinalang', name: 'ballerina-runtime', version: "${ballerinaLangVersion}"
+//
+//        compile group: 'com.sun.mail', name: 'javax.mail', version: "${javaMailVersion}"
+//        compile group: 'com.icegreen', name: 'greenmail', version: "${greenmailVersion}"
+//        compile group: 'com.sun.activation', name: 'jakarta.activation', version: "${jakartaActivationVersion}"
+//        compile group: 'org.testng', name: 'testng', version: "${testngVersion}"
+//        compile group: 'org.jvnet.mimepull', name: 'mimepull', version: "${mimepullVersion}"
```
Comment (kalaiyarasiganeshalingam) [inline build.gradle:178]: Can we remove these If not needed?

File: email-native/src/main/java/module-info.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    exports org.ballerinalang.stdlib.email.util;
+    exports org.ballerinalang.stdlib.email.client;
+    exports org.ballerinalang.stdlib.email.service.compiler;
+    requires io.ballerina.jvm;
+    requires org.slf4j;
+    requires io.ballerina.lang;
+    requires java.mail;
+    requires io.ballerina.stdlib.mime;
+    requires io.ballerina.stdlib.io;
+}
```
Comment (ThisaruGuruge) [inline email-native/src/main/java/module-info.java:11]:
```suggestion
}

```

File: email-native/src/main/java/module-info.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+module io.ballerina.stdlib.email {
```
Comment (ThisaruGuruge) [inline email-native/src/main/java/module-info.java:1]: Missing license headers

File: build.gradle [hunk 5]
```diff
@@ -38,6 +38,7 @@ ext.testngVersion = "6.14.3"
 ext.slf4jVersion = "1.7.30"
 ext.javaMailVersion = "1.6.2"
 ext.greenmailVersion = "1.5.11"
+//ext.jakartaActivationVersion = "1.2.2"
```
Comment (ThisaruGuruge) [inline build.gradle:41]: Shall we remove this if we don't use it anymore?

File: build.gradle [hunk 6]
```diff
@@ -69,9 +70,13 @@ allprojects {
...
-        maven {
-            url = 'https://mvnrepository.com/artifact/com.icegreen/greenmail'
-        }
+//        maven {
+//            url = 'https://mvnrepository.com/artifact/com.icegreen/greenmail'
+//        }
+
+//        maven {
+//            url = 'https://mvnrepository.com/artifact/com.sun.activation/javax.activation'
+//        }
```
Comment (ThisaruGuruge) [inline build.gradle:79]: Shall we remove the commented urls? Check other places as well.

**Review by ThisaruGuruge — APPROVED** [review]

File: email-test-utils/src/main/java/module-info.java (hunk 1 above)
Comment (Maninda) [reply email-test-utils/src/main/java/module-info.java:19]: Added.

File: build.gradle (hunk 2 above)
Comment (Maninda) [reply build.gradle]: Removed.

File: email-native/src/main/java/module-info.java (hunk 3 above)
Comment (Maninda) [reply email-native/src/main/java/module-info.java]: Added the given changes with my commit.

File: build.gradle (hunk 5 above)
Comment (Maninda) [reply build.gradle]: Removed.

File: build.gradle (hunk 6 above)
Comment (Maninda) [reply build.gradle]: Removed.

File: email-native/src/main/java/module-info.java (hunk 4 above)
Comment (Maninda) [reply email-native/src/main/java/module-info.java:19]: Added.

**Pull request merged**